jobrunner/executors/local.py

```diff
@@ -72,17 +72,34 @@ def get_log_dir(job_id):
     return config.JOB_LOG_DIR / month_dir / container_name(job_id)
 
 
-def read_job_metadata(job_id):
+def read_job_metadata(job_id, redact=True):
     path = job_metadata_path(job_id)
     if path:
         metadata = json.loads(path.read_text())
         # ensure all metadata has default values, as older json files might be
         # missing some
-        return METADATA_DEFAULTS | metadata
-
+        metadata = METADATA_DEFAULTS | metadata
+        if redact:
+            redact_job_metadata(metadata)
+        return metadata
     return {}
 
 
+def redact_job_metadata(metadata):
+    """
+    Redact output filenames and patterns from the job metadata
+    If there are unmatched outputs or patterns, we also redact the
+    status_message and hint which may contain messages referring to
+    the unmatched filenames or patterns.
+    """
+    for key in ["unmatched_outputs", "unmatched_patterns"]:
+        if metadata[key]:
+            metadata["status_message"] = ""
+            metadata["hint"] = ""
+        del metadata[key]
+    del metadata["outputs"]
```

Comment:
My suggested code above relies on it not being an error to redact an empty dict, but I think that better captures the behaviour we want here. Using `metadata.pop(key,  None)` instead of `del metadata[key]` would do that.